### PR TITLE
implemented User and Role models, security

### PR DESCRIPTION
**Pull request by mmosii**

### CODE REVIEW

**Review by teract10s — APPROVED** [review]
LGTM. I don't know what can i add

File: src/main/java/mmosii/bookstore/repository/user/RoleRepository.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package mmosii.bookstore.repository.user;
+
+import mmosii.bookstore.model.Role;
+import mmosii.bookstore.model.RoleName;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface RoleRepository extends JpaRepository<Role, Long> {
+    Role findAllByName(RoleName name);
```
Comment (Elena-Bruyako) [inline src/main/java/mmosii/bookstore/repository/user/RoleRepository.java]: It's better findRoleByName or getRoleByName

File: src/main/java/mmosii/bookstore/security/CustomUserDetailsService.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+@Component
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        return userRepository.findByEmail(username).orElseThrow(()
+                -> new UsernameNotFoundException("Can't find user"));
```
Comment (Elena-Bruyako) [inline src/main/java/mmosii/bookstore/security/CustomUserDetailsService.java]: Also, you can add username 
"Can't find user by username: " + username

File: src/main/java/mmosii/bookstore/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            UserDetails userDetails = userDetailsService.loadUserByUsername(username);
+            Authentication authentication = new UsernamePasswordAuthenticationToken(userDetails,
+                    null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/mmosii/bookstore/security/JwtAuthenticationFilter.java]: it's better authorizationHeader by meaning

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's fix some comments.

**Review by Andriy-dz — APPROVED** [review]

**Pull request merged**